src/pwm.rs

```diff
@@ -258,10 +300,95 @@ macro_rules! hal {
                         .set_bit()
                 );
 
-                unsafe { mem::MaybeUninit::uninit().assume_init() }
+                Pwm {
+                    clk,
+                    _pins: PhantomData,
+                }
+            }
+
+            impl<REMAP, P, PINS> hal::Pwm for Pwm<$TIMX, REMAP, P, PINS> where
+                REMAP: Remap<Periph = $TIMX>,
+                PINS: Pins<REMAP, P>,
+            {
+                type Channel = Channel;
+                type Duty = u16;
+                type Time = Hertz;
+
+                fn enable(&mut self, channel: Self::Channel) {
+                    match PINS::check_enabled(channel) {
+                        Some(Channel::C1) => unsafe { bb::set(&(*$TIMX::ptr()).ccer, 0) },
+                        Some(Channel::C2) => unsafe { bb::set(&(*$TIMX::ptr()).ccer, 4) },
+                        Some(Channel::C3) => unsafe { bb::set(&(*$TIMX::ptr()).ccer, 8) },
+                        Some(Channel::C4) => unsafe { bb::set(&(*$TIMX::ptr()).ccer, 12) },
+                        None => {}
+                    }
+                }
+
+                fn disable(&mut self, channel: Self::Channel) {
+                    match PINS::check_enabled(channel) {
+                        Some(Channel::C1) => unsafe { bb::clear(&(*$TIMX::ptr()).ccer, 0) },
+                        Some(Channel::C2) => unsafe { bb::clear(&(*$TIMX::ptr()).ccer, 4) },
+                        Some(Channel::C3) => unsafe { bb::clear(&(*$TIMX::ptr()).ccer, 8) },
+                        Some(Channel::C4) => unsafe { bb::clear(&(*$TIMX::ptr()).ccer, 12) },
+                        None => {}
+                    }
+                }
+
+                fn get_duty(&self, channel: Self::Channel) -> Self::Duty {
+                    match PINS::check_enabled(channel) {
+                        Some(Channel::C1) => unsafe { (*$TIMX::ptr()).ccr1.read().ccr().bits() },
+                        Some(Channel::C2) => unsafe { (*$TIMX::ptr()).ccr2.read().ccr().bits() },
+                        Some(Channel::C3) => unsafe { (*$TIMX::ptr()).ccr3.read().ccr().bits() },
+                        Some(Channel::C4) => unsafe { (*$TIMX::ptr()).ccr4.read().ccr().bits() },
+                        None => 0,
```

Comment:
Is 0 always equivalent to a disabled channel? Or can the channel be in some sort of floating state if disabled?

Reply:
What about to panic here?

Reply:
I would really like to avoid panics as far as possible. I don't see a great way to accomplish that though :/

Reply:
By experience, disable is different to duty of 0

Reply:
But it should be OK to give the duty whatever it is enabled or disabled, according to enabling the channel would give you this duty.

Reply:
Yep, that seems like the best solution